app/frontend/src/features/search/FilterDialog.tsx

```diff
@@ -0,0 +1,191 @@
+      aria-labelledby="filter-dialog-title"
+    >
+      <DialogTitle id="filter-dialog-title">{FILTER_DIALOG_TITLE}</DialogTitle>
+      <form onSubmit={onSubmit}>
+        <DialogContent>
+          <LocationAutocomplete
+            control={control}
+            defaultValue={
+              searchParams.has("location") &&
+              searchParams.has("lng") &&
+              searchParams.has("lat")
+                ? {
+                    name: searchParams.get("location")!,
+                    simplifiedName: searchParams.get("location")!,
+                    location: new LngLat(
+                      Number.parseFloat(searchParams.get("lng")!) || 0,
+                      Number.parseFloat(searchParams.get("lat")!) || 0
+                    ),
+                  }
+                : undefined
+            }
+            onChange={(value) => {
+              if (value === "") {
+                searchParams.delete("location");
+                searchParams.delete("lat");
+                searchParams.delete("lng");
+              } else {
+                searchParams.set("location", value.simplifiedName);
+                searchParams.set("lat", value.location.lat.toString());
+                searchParams.set("lng", value.location.lng.toString());
+              }
+            }}
+          />
+          <Divider />
+          <Grid container spacing={2}>
+            <Grid item xs={12} md={6} className={classes.container}>
+              <Typography variant="h3">{HOST_FILTERS}</Typography>
+              <Autocomplete
+                id="last-active-filter"
+                label={LAST_ACTIVE}
+                options={lastActiveOptions}
+                getOptionLabel={(o) => o.label}
+                onChange={(_e, option) =>
+                  option
+                    ? searchParams.set("lastActive", option.value.toString())
+                    : searchParams.delete("lastActive")
+                }
+                defaultValue={lastActiveOptions.find(
+                  (o) => o.value.toString() === searchParams.get("lastActive")
+                )}
+                disableClearable={false}
+                freeSolo={false}
+                multiple={false}
+              />
+              <Autocomplete<HostingStatus, true, false, false>
+                id="host-status-filter"
+                label={HOSTING_STATUS}
+                options={hostingStatusOptions}
+                onChange={(_e, options) =>
+                  options && options.length !== 0
+                    ? searchParams.set("hostingStatus", options.join(","))
+                    : searchParams.delete("hostingStatus")
+                }
+                defaultValue={
+                  searchParams.has("hostingStatus")
+                    ? searchParams
+                        .get("hostingStatus")
+                        ?.split(",")
+                        .map((k) => Number.parseInt(k))
+                    : undefined
```

Comment:
Since search parameters can be repeated (i.e. `?hostingStatus=1&hostingStatus=2` is valid) and `URLSearchParams` pick that up correctly, it might be worth using its API more by doing something like this in the `onChange` for the bit that's setting `hostingStatus`:
```typescript
options.forEach((option, index) => {
  if (index === 0) {
    searchParams.set("hostingStatus", option);
  } else {
    searchParams.append("hostingStatus", option);
  }
});
```

So then in `defaultValue` and in `SearchPage`, you can do `searchParams.getAll("hostingStatus")?.map((v) => Number(v))` without needing to split